### ISSUE TITLE
running error

### ISSUE THREAD

**Issue by wenzhengchen** [BachiLi/redner#9]
Hi, 

When I run your tutorial/01_optimize_single_triangle.py, when the code run line 146 loss.backward()

It has an Assertion error.

/h/wenzheng/pydr2/lib/python3.7/site-packages/skimage/util/dtype.py:141: UserWarning: Possible precision loss when converting from float32 to uint8
  .format(dtypeobj_in, dtypeobj_out))
iteration: 0
loss: 758.4083251953125
grad: tensor([[-131.9577,  582.0557, -426.4650],
        [ 183.6868,  882.2390, -330.1663],
        [  58.8177, -528.9055, -201.2860]])
vertices: tensor([[-1.9500,  1.4500,  0.3500],
        [ 0.8500,  1.1500, -0.2500],
        [-0.4500, -1.3500,  0.2500]], requires_grad=True)
iteration: 1
loss: 609.8726196289062
python: /h/wenzheng/project/redner/material.h:208: Vector3 bsdf(const Material&, const SurfacePoint&, const Vector3&, const Vector3&, Real): Assertion `roughness > 0.f' failed.
python: /h/wenzheng/project/redner/material.h:208: Vector3 bsdf(const Material&, const SurfacePoint&, const Vector3&, const Vector3&, Real): Assertion `roughness > 0.f' failed.
python: /h/wenzheng/project/redner/material.h:208: Vector3 bsdf(const Material&, const SurfacePoint&, const Vector3&, const Vector3&, Real): Assertion `roughness > 0.f' failed.
Aborted (core dumped)

Could you please tell me how to fix this?

Thank you!

**Comment by wenzhengchen** [comment]
I compile this code with just CPU.

**Comment by BachiLi** [comment]
Hi,

Unfortunately I can't reproduce this. Do you still have this issue after pulling from the current latest commit? If so, can you try to print out roughness in the backward() function in render_pytorch.py?

**Comment by BachiLi** [comment]
It might be caused by a confusion between 0-dimensional and 1-dimensional tensor. Can you pull again from the latest commit and see if it still happens?
